tests/segarray_test.py

```diff
@@ -436,26 +436,111 @@ def test_remove_repeats(self):
+        self.assertListEqual(dedup[3].to_ndarray().tolist(), list(set(b)))
+        self.assertListEqual(dedup[4].to_ndarray().tolist(), [])
+
+    def test_intersection(self):
+        a = [1, 2, 3, 4, 5]
+        b = [6, 7, 8]
+        c = [1, 2, 4]
+        d = [8]
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a + b))
+        segarr_2 = ak.SegArray(ak.array([0, len(c)]), ak.array(c + d))
+
+        intx = segarr.intersect(segarr_2)
+
+        self.assertEqual(intx.size, 2)
+        self.assertListEqual(intx[0].to_ndarray().tolist(), [1, 2, 4])
+        self.assertListEqual(intx[1].to_ndarray().tolist(), [8])
+
+        #test with empty Segments
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a))
+        intx = segarr.intersect(segarr_2)
+        self.assertListEqual(intx.lengths.to_ndarray().tolist(), [3, 0])
+        self.assertListEqual(intx[0].to_ndarray().tolist(), [1, 2, 4])
+        self.assertListEqual(intx[1].to_ndarray().tolist(), [])
+
+
+    def test_union(self):
+        a = [1, 2, 3, 4, 5]
+        b = [6, 7, 8]
+        c = [1, 2, 4]
+        d = [8]
+
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a + b))
+        segarr_2 = ak.SegArray(ak.array([0, len(c)]), ak.array(c + d))
+
+        un = segarr.union(segarr_2)
+        self.assertEqual(un.size, 2)
+        self.assertListEqual(un[0].to_ndarray().tolist(), [1, 2, 3, 4, 5])
+        self.assertListEqual(un[1].to_ndarray().tolist(), [6, 7, 8])
+
+        #test with empty segments
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a))
+        un = segarr.union(segarr_2)
+        self.assertListEqual(un.lengths.to_ndarray().tolist(), [5, 1])
+        self.assertListEqual(un[0].to_ndarray().tolist(), [1, 2, 3, 4, 5])
+        self.assertListEqual(un[1].to_ndarray().tolist(), [8])
+
+    def test_difference(self):
+        a = [1, 2, 3, 4, 5]
+        b = [6, 7, 8]
+        c = [1, 2, 4]
+        d = [8]
+
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a + b))
+        segarr_2 = ak.SegArray(ak.array([0, len(c)]), ak.array(c + d))
+
+        diff = segarr.difference(segarr_2)
+        self.assertEqual(diff.size, 2)
+        self.assertListEqual(diff[0].to_ndarray().tolist(), [3, 5])
+        self.assertListEqual(diff[1].to_ndarray().tolist(), [6, 7])
+
+        # test with empty segments
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a))
+        diff = segarr.difference(segarr_2)
+        self.assertListEqual(diff.lengths.to_ndarray().tolist(), [2, 0])
+        self.assertListEqual(diff[0].to_ndarray().tolist(), [3, 5])
+        self.assertListEqual(diff[1].to_ndarray().tolist(), [])
+
+    def test_xor(self):
+        a = [1, 2, 3]
+        b = [6, 7, 8]
+        c = [1, 2, 4]
+        d = [8, 12, 13]
+
+        segarr = ak.SegArray(ak.array([0, len(a)]), ak.array(a + b))
+        segarr_2 = ak.SegArray(ak.array([0, len(c)]), ak.array(c + d))
+        diff = segarr.xor(segarr_2)
```

Comment:
not super important but it might be nice to rename this to xor to avoid confusion with the above test

Reply:
Updated variable name